server/routes/users.js

```diff
@@ -1,12 +1,15 @@
 import { Router } from 'express';
 import User from '../controllers/user';
 import inputValidator from '../middlewares/validations';
-import checkAuth from '../middlewares/checkAuth';
+import isAuthenticated from '../middlewares/checkAuth';
 
 const router = new Router();
 
 router.post('/users/signup', inputValidator.signupValidator, User.signup);
 router.post('/users/signin', inputValidator.signinValidator, User.signin);
-router.get('/users', checkAuth, User.listAll);
+router.get('/users', isAuthenticated, User.listAll);
+router.post('/users/resetpassword', User.resetPassword);
```

Comment:
Shouldn't you check to ensure that the currently logged in user is only reseting their own password?

Reply:
This can be handled on the front end with profile reducer states. So upon logging in if a user wishes to reset their password then we send the request on his behalf with the signed email address. If I authenticate the route before handling the reset password then a user who forgot their password and isn't signed in won't be able to reset their password.